pkg/config/initoptions.go

```diff
@@ -115,34 +116,37 @@ func isDirectory(path string) bool {
 	return false
 }
 
-// packageNamespace returns the namespace of the package
-// config files.
-func packageNamespace(packageDir string) (string, error) {
+// calcPackageNamespace returns the namespace of the package
+// config files. Assumes all namespaced resources are in the
+// same namespace. Returns the default namespace if none of the
+// config files has a namespace.
+func calcPackageNamespace(packageDir string) (string, error) {
 	r := kio.LocalPackageReader{PackagePath: packageDir}
 	nodes, err := r.Read()
 	if err != nil {
 		return "", err
 	}
-	namespace := metav1.NamespaceDefault
+	// Return the first non-empty namespace found. Cluster-scoped
+	// resources do not have namespace set.
 	for _, node := range nodes {
 		rm, err := node.GetMeta()
 		if err != nil {
 			continue
 		}
 		if len(rm.ObjectMeta.Namespace) > 0 {
-			namespace = rm.ObjectMeta.Namespace
+			return rm.ObjectMeta.Namespace, nil
 		}
-		break
 	}
-	return namespace, nil
+	// Return the default namespace if none found.
+	return metav1.NamespaceDefault, nil
```

Comment:
Maybe we should just ask the user to tell us which namespace in this case rather than just taking the default namespace? Or we could ask the user to confirm the choices as part of the `init` command?

Reply:
I think we should definitely put more thought into a solution. But this fixes a bug. Implementing a system for user input, and then validating (currently we only allow one namespace) would be a bigger effort. Let's get this fix in, and implement the bigger solution in another PR.